Add gas cost for `UnpackAny`

chipshort opened the pull request:
breaking for people using a custom Any encoder

Benchmark results:
```
BenchmarkUnpackAny/garbage_any-8          2235267       521.9 ns/op     136 B/op       4 allocs/op
BenchmarkUnpackAny/single_MsgExec-8       1736608       686.1 ns/op       24 B/op       2 allocs/op
BenchmarkUnpackAny/10000_MsgExec-8        1724133       689.4 ns/op       24 B/op       2 allocs/op
BenchmarkUnpackAny/100000_MsgExec-8       1723904       690.6 ns/op       24 B/op       2 allocs/op
```

So, it looks like it takes roughly 700ns
Our gas target in CosmWasm uses a target of 1e12 Gas/s = 1e3 Gas/ns. That means:
1e3 Gas/ns * 700ns = 700000 Gas (in CW Gas units)

With the default gas multiplier, it's 5 SDK gas

pinosu commented on x/wasm/keeper/handler_plugin_encoders.go:
```diff
@@ -206,12 +206,14 @@ func EncodeStakingMsg(sender sdk.AccAddress, msg *wasmvmtypes.StakingMsg) ([]sdk
 		}
 		var sdkMsg sdk.Msg
+
+		ctx.GasMeter().ConsumeGas((700000*uint64(len(msg.Value)))/types.DefaultGasMultiplier, "unpacking AnyMsg")
```
what about extracting 700000 as a const and adding a short comment about what it represents (similar to what you did in the pr description)

chipshort replied:
Nice idea! Will do that

pinosu reviewed (approved):
Great work! LGTM 👍

chipshort commented:
Ok, I added the constant, but also fixed a bug that I forgot to push the fix for: The gas cost (at least according to my benchmark) seems pretty constant and not related to the message size (see numbers above), so it should not be multiplied with the length
